src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -480,6 +480,21 @@ public void AddResultsForTarget(string target, TargetResult result)
             _resultsByTarget[target] = result;
         }
 
+        /// <summary>
+        /// Keep the results for specific targets only.
+        /// </summary>
+        /// <param name="targets">The targets whose results should be kept.</param>
+        public void KeepResultsForSpecificTargets(HashSet<string> targets)
```

Comment:
This should probably be `internal` to avoid exposing additional public surface area